process/logsevents/logsmerge.go

```diff
@@ -0,0 +1,73 @@
+package logsevents
+
+import (
+	"encoding/hex"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+)
+
+var log = logger.GetOrCreate("process/logsevents")
+
+type logsMerger struct {
+	hasher      hashing.Hasher
+	marshalizer marshal.Marshalizer
+}
+
+func NewLogsMerger(hasher hashing.Hasher, marshalizer marshal.Marshalizer) (*logsMerger, error) {
+	if check.IfNil(hasher) {
+		return nil, ErrNilHasher
+	}
+	if check.IfNil(marshalizer) {
+		return nil, ErrNilMarshalizer
+	}
+
+	return &logsMerger{
+		hasher:      hasher,
+		marshalizer: marshalizer,
+	}, nil
+}
+
+func (lm *logsMerger) MergeLogEvents(logSource *transaction.ApiLogs, logDestination *transaction.ApiLogs) *transaction.ApiLogs {
+	if logSource == nil {
```

Comment:
what if both are nil? it will return `logDestination` which is `nil`. is this expected?

Reply:
yes. this is expected behavior because if both logs are nil the result should be also nil.